### ISSUE TITLE
WARNING [step vecscreen] completed permanentFail when running GCA_000166555

### ISSUE THREAD

**Issue by AndreinaCastillo** [ncbi/pgap#33]
Hello,

I wanted to inquire about some issues I have been having when using PGAP on fasta files containing multiple >headers for the same organism. I have tried running the pipeline using both my own input data and the sample file GCA_000166555 from the test_genome folder. In both cases, the analysis always fails during the same step with the following error message:

[2019-08-19 18:12:48] INFO Could not collect memory usage, job ended before monitoring began.
[2019-08-19 18:12:48] WARNING [job screen_evaluate] completed permanentFail
[2019-08-19 18:12:48] WARNING [step screen_evaluate] completed permanentFail
[2019-08-19 18:12:48] INFO [workflow vecscreen] completed permanentFail
[2019-08-19 18:12:48] WARNING [step vecscreen] completed permanentFail
[2019-08-19 18:12:48] INFO [workflow ] completed permanentFail

I also tried running the analysis using the test files from MG37 and the pipeline seemed to run without issues in that case. 

Any ideas how can I solve this issue or what I might be doing incorrectly? I can provide the complete cwltool.log file as well as the input files used if needed. 

Thanks in advance!

**Comment by tseemann** [comment]
Was this on Linux or MacOS ?

**Comment by azat-badretdin** [comment]
First of all, apologies for not replying sooner.

>Any ideas how can I solve this issue or what I might be doing incorrectly? 

Most likely, the reason is that current screening software has a tiny problem: in the end of the execution of `vecscreen.cwl` workflow we are testing for the presence of non-empty containers in two files: `calls.tab` and `feats.asn`. The former is actual final file with features on nucleotide sequences showing contaminated regions. If it is not empty, then, indeed, it means that we detected some contamination. The latter is actually an intermediate result, obtained before we filter out what we think are false positives. It was added as input to the final "decider node" (`screen_evaluate.cwl`) by our mistake. Good news is that we already fixed that bug, but `origin/prod` branch of our software still has this bug (and this is what you are using now, probably), while `origin/test` branch already does not, judging by the code of `vecscreen.cwl`.

Bottom line, it does not seem like you made a mistake.

**Comment by AndreinaCastillo** [comment]
Hello,

Sorry for the late reply. tseemann, I am using an AWS Linux AMI 2018.03.0 (HVM) EC2 instance.
azat-badretdin, that makes sense. Thanks for letting me know! Since I followed the install instructions listed on (https://github.com/ncbi/pgap/wiki/Quick-Start), should I try to download and re-install the pipeline from the origin/test branch, or would it be better to wait for the bug fix to reach the origin/prod branch and then reinstall?

Thanks to both for the feedback! 

Best,
Andreina

**Comment by azat-badretdin** [comment]
>should I try to download and re-install the pipeline from the origin/test branch

Feel free to try `origin/test` if you expect that you might have borderline genomes like the one from our panel. And if you are sure that your genomes are not likely to be contaminated, keep running them with the official version from Quick-Start.

**Comment by AndreinaCastillo** [comment]
I downloaded the code from https://github.com/ncbi/pgap/tree/test/scripts/ and I am still getting the same error message. The error remains either with GCA_000166555 or our own genomes. Am I in using the correct branch? 
Alternatively, would running <./pgap.py --update> lead to download files from the official version instead of origin/test? 
Thanks again!

**Comment by azat-badretdin** [comment]
`./pgap.py --update` will probably download it from the official version. 

Could you please post the sequence of the shell commands you ran to download the code and then run it? And the head and tail of `cwltool.log` like you did before, please? Thanks!

**Comment by AndreinaCastillo** [comment]
The sequence of shell commands was:
wget https://raw.githubusercontent.com/ncbi/pgap/test/scripts/pgap.py
chmod +x pgap.py
./pgap.py --update
./pgap.py -r -o GCA_000166555_results test_genomes/GCA_000166555/input.yaml

The head of cwltool.log:
Original command: ./pgap.py -r -o GCA_000166555_results test_genomes/GCA_000166555/input.yaml

Docker command: /usr/bin/docker run -i --user 500:500 --volume /home/ec2-user/branch/input-2019-08-01.build3919:/pgap/input:ro,z --volume /home/ec2-user/branch/test_genomes/GCA_000166555:/pgap/user_input:z --volume /home/ec2-user/branch/test_genomes/GCA_000166555/pgap_input.yaml:/pgap/user_input/pgap_input.yaml:ro,z --volume /home/ec2-user/branch/GCA_000166555_results:/pgap/output:rw,z ncbi/pgap:2019-08-01.build3919 cwltool --timestamps --disable-color --outdir /pgap/output pgap.cwl /pgap/user_input/pgap_input.yaml

--- Start YAML Input ---
fasta:
  class: File
  location: GCA_000166555.2_ASM16655v2_genomic.fasta
submol:
  class: File

The tail of cwtool.log:
[2019-08-21 20:01:17] WARNING Final process status is permanentFail
[ec2-user@ip-172-31-22-146 GCA_000166555_results]$ tail -n 30 cwltool.log
[2019-08-21 20:01:17] INFO Could not collect memory usage, job ended before monitoring began.
[2019-08-21 20:01:17] WARNING [job screen_evaluate] completed permanentFail
[2019-08-21 20:01:17] WARNING [step screen_evaluate] completed permanentFail
[2019-08-21 20:01:17] INFO [workflow vecscreen] completed permanentFail
[2019-08-21 20:01:17] WARNING [step vecscreen] completed permanentFail
[2019-08-21 20:01:17] INFO [workflow ] completed permanentFail
{
    "gbk": null,
    "gff": null,
    "input_fasta": {
        "class": "File",
        "location": "file:///pgap/output/GCA_000166555.2_ASM16655v2_genomic.fasta",
        "size": 5795751,
        "basename": "GCA_000166555.2_ASM16655v2_genomic.fasta",
        "checksum": "sha1$8da4844328aa7d292d323f9cbb9545282399e419",
        "path": "/pgap/output/GCA_000166555.2_ASM16655v2_genomic.fasta"
    },
    "input_submol": {
        "class": "File",
        "location": "file:///pgap/output/submol.yaml",
        "size": 1521,
        "basename": "submol.yaml",
        "checksum": "sha1$3b5f229166fe758f1b97b37f28225503b8f446b4",
        "path": "/pgap/output/submol.yaml"
    },
    "nucleotide_fasta": null,
    "protein_fasta": null,
    "sqn": null
}
[2019-08-21 20:01:17] WARNING Final process status is permanentFail

Thanks!

**Comment by azat-badretdin** [comment]
Thanks for quick reply! Judging by your shell `history` it looks like you did not run the test version.

**Comment by AndreinaCastillo** [comment]
Thanks to you as well! 
Well, that makes sense. I will try again without running the --update command.

**Comment by azat-badretdin** [comment]
I am also running this  [locally, internal link, don't click](https://teamcity.ncbi.nlm.nih.gov/viewQueued.html?itemId=23492325&tab=queuedBuildOverviewTab) to verify that we indeed do not have *vecscreen* problems with that. The software branch: **origin/test**

**Comment by azat-badretdin** [comment]
> I am also running this locally, internal link, don't click to verify that we indeed do not have vecscreen problems with that. The software branch: origin/test

That has been verified. The code was merged to `origin/prod`

**Comment by AndreinaCastillo** [comment]
Thanks for the update. I will try  to install PGAP again following the https://github.com/ncbi/pgap/wiki/Quick-Start instructions. I will let how it works as soon as possible.

**Comment by AndreinaCastillo** [comment]
Just finished re-installing the pipeline and running the command  ./pgap.py -r -o GCA_000166555_results test_genomes/GCA_000166555/input.yaml

I am afraid I am still getting the error. I am including the tail of cwtool.log:
[2019-08-22 19:08:58] INFO [job fscr_calls_pass1] Max memory used: 36MiB
[2019-08-22 19:08:58] INFO [job fscr_calls_pass1] completed success
[2019-08-22 19:08:58] INFO [step fscr_calls_pass1] completed success
[2019-08-22 19:08:58] INFO [workflow FSCR_Calls_first_pass] starting step fscr_format_calls
[2019-08-22 19:08:58] INFO [step fscr_format_calls] start
[2019-08-22 19:08:58] INFO [job fscr_format_calls] /tmp/v9n9qids$ fscr_format_calls \
    -calls \
    /tmp/tmp2_cfj8sx/stg9c5b00f6-f98f-43fe-ac13-85480aba2722/calls.asn \
    -o \
    calls.tab
Error: (308.5) [some bogus value]  Service not found
Error: (315.2) CConn_Streambuf::CConn_Streambuf():  NULL connector: Unknown
Error: (802.5) CObjectOStream: error at byte 6: stack is empty: write fault
[2019-08-22 19:09:00] INFO [job fscr_format_calls] Max memory used: 35MiB
[2019-08-22 19:09:00] INFO [job fscr_format_calls] completed success
[2019-08-22 19:09:00] INFO [step fscr_format_calls] completed success
[2019-08-22 19:09:00] INFO [workflow FSCR_Calls_first_pass] completed success
[2019-08-22 19:09:00] INFO [step FSCR_Calls_first_pass] completed success
[2019-08-22 19:09:00] INFO [workflow vecscreen] starting step screen_evaluate
[2019-08-22 19:09:00] INFO [step screen_evaluate] start
[2019-08-22 19:09:00] INFO [job screen_evaluate] /tmp/h760b1_i$ screen_evaluate \
    -asn \
    /tmp/tmpdqje3kk1/stg264350c5-4882-4474-a1ea-d5e2c9570602/feats.asn \
    -ifmt \
    seq-annot \
    -tab \
    /tmp/tmpdqje3kk1/stgf8717e93-8578-4461-b65e-e2cd7a73a9df/calls.tab
[2019-08-22 19:09:00] INFO Could not collect memory usage, job ended before monitoring began.
[2019-08-22 19:09:00] WARNING [job screen_evaluate] completed permanentFail
[2019-08-22 19:09:00] WARNING [step screen_evaluate] completed permanentFail
[2019-08-22 19:09:00] INFO [workflow vecscreen] completed permanentFail
[2019-08-22 19:09:00] WARNING [step vecscreen] completed permanentFail
[2019-08-22 19:09:00] INFO [workflow ] completed permanentFail
{
    "gbk": null,
    "gff": null,
    "input_fasta": {
        "class": "File",
        "location": "file:///pgap/output/GCA_000166555.2_ASM16655v2_genomic.fasta",
        "size": 5795751,
        "basename": "GCA_000166555.2_ASM16655v2_genomic.fasta",
        "checksum": "sha1$8da4844328aa7d292d323f9cbb9545282399e419",
        "path": "/pgap/output/GCA_000166555.2_ASM16655v2_genomic.fasta"
    },
    "input_submol": {
        "class": "File",
        "location": "file:///pgap/output/submol.yaml",
        "size": 1521,
        "basename": "submol.yaml",
        "checksum": "sha1$3b5f229166fe758f1b97b37f28225503b8f446b4",
        "path": "/pgap/output/submol.yaml"
    },
    "nucleotide_fasta": null,
    "protein_fasta": null,
    "sqn": null
}
[2019-08-22 19:09:00] WARNING Final process status is permanentFail

The installed version is still input-2019-08-01.build3919.

My apologies, I am not sure what I am doing wrong.

**Comment by azat-badretdin** [comment]
> Thanks for the update. I will try to install PGAP again following the https://github.com/ncbi/pgap/wiki/Quick-Start instructions. I will let how it works as soon as possible

>The installed version is still input-2019-08-01.build3919

> My apologies, I am not sure what I am doing wrong

You have nothing to apologize for. We need to do something more on our side first. I have an open internal ticket for this. Currently I am running a complete test panel with `origin/prod` code, then I will run a TeamCity build that will create a release. 

We will notify you once we have it ready.

**Comment by AndreinaCastillo** [comment]
Thanks!

**Comment by AndreinaCastillo** [comment]
Got it. Thanks!
